### ISSUE TITLE
Support remaining commands which samsung-lfd handles

### ISSUE THREAD

**Issue by peternewman** [bitfocus/companion-module-samsung-display#10]
After https://github.com/bitfocus/companion-module-samsung-display/pull/8 is in, it would be good to add the remaining commands (mostly video wall) that are already handled by the library (see samsungd.xml of https://www.npmjs.com/package/samsung-lfd?activeTab=code ).

Potentially also other commands but that would involve updating the NPM module itself.

**Comment by peternewman** [comment]
Progress on this in #8 but I think e.g. video wall switching hasn't been added yet.

**Comment by HughBourne** [comment]
Do you have the latest MDC document from samsung? If you need it please let me know. I work for samsung on the pro display side.  We also have  a range of LED products used in the broadcast space that use MDC and it would be nice to see this added to companion etc.

**Comment by peternewman** [comment]
> Do you have the latest MDC document from samsung? If you need it please let me know. I work for samsung on the pro display side.

I think the last one I found online was from the late 2010s, so if you're able to attach a newer one to this issue would be great thanks @HughBourne !

I think the other part of the challenge, if you/your colleagues are able to do anything about it, is the data parsing side of things. For example https://mdc.sesg.ch/ seems to have 125 odd commands, some of which have 5-10 arguments to them, that's a hell of a lot of info to try and type in accurately. Whereas if Samsung could deliver a YAML or a JSON or whatever with the parameters, it would make it so much easier to implement more of the commands in https://github.com/magik1969/samsung-lfd that drives this particular Companion plugin, or https://github.com/vgavro/samsung-mdc or https://github.com/magik1969/samsung-mdc-protocol or some of the many other app implementations.

> We also have a range of LED products used in the broadcast space that use MDC and it would be nice to see this added to companion etc.

Do they need a separate module/connection (or tweaks to this one), or do they already work with this existing Samsung Display module?

**Comment by HughBourne** [comment]
@peternewman thanks for the speedy response. The current version of the doc is v15 and is a pdf. It’s pretty big. I recently got Claude to convert it to a .md file for a project I was working on. I will have a look at providing an xml as I think that would make it a lot easier than JSON for reading. 

All of the LFD products support MDC but there are a few LED specific commands.  Having a specific module for LED might be a good idea otherwise it would require a fair bit of logic in the module to work out what device is connected.

**Comment by peternewman** [comment]
> [@peternewman](https://github.com/peternewman) thanks for the speedy response. The current version of the doc is v15 and is a pdf. It’s pretty big.

It looks like GitHub allows up to a 25MB document to be attached (or 25MB of .zip, .gz or .tgz if that helps shrink it):
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files

Although if v15 is the latest, it seems to be here:
https://vgavro.github.io/samsung-mdc/MDC-Protocol.pdf

> I recently got Claude to convert it to a .md file for a project I was working on. I will have a look at providing an xml as I think that would make it a lot easier than JSON for reading.

Yeah XML is also obviously consumable fine. Although it may be as easy to generate some other format from what the Python people have already processed if you're going to have to point AI at it. However they've "only" implemented 81 of the 125 commands the other website lists (which is probably still less than the 2020 v15 spec has!

> All of the LFD products support MDC but there are a few LED specific commands. Having a specific module for LED might be a good idea otherwise it would require a fair bit of logic in the module to work out what device is connected.

I assume that's the cabinet type stuff is it? But presumably they support the general brightness/contrast type commands already?

Are there lots of "TV" commands they don't support on the cabinet?

Having everything in samsung-lfd and then different Companion modules wouldn't be too hard, but would probably still be a chunk of duplication. The other way to do it, would be a config option in the module to enable LED stuff, or switch between LED and "TV" or whatever, to then show/hide the unnecessary commands.

**Comment by HughBourne** [comment]
Yeah that’s pretty close to the latest. The led specific commands are from page 256 of that doc. To be fair there are a lot of commands that aren’t really needed unless you want complete control over everything. 
I did notice that the hdmi1, hdmi source command doesn’t work. I haven’t had time to dig into it though. 
I will see about getting something done that puts all the commands into a sensible format. Hopefully next week if I get time.
